exp/controllers/azuremachinepoolmachine_controller.go

```diff
@@ -186,27 +186,57 @@ func (ampmr *AzureMachinePoolMachineController) Reconcile(ctx context.Context, r
 		return ctrl.Result{}, nil
 	}
 
-	azureClusterName := client.ObjectKey{
-		Namespace: machine.Namespace,
-		Name:      cluster.Spec.InfrastructureRef.Name,
-	}
+	var clusterScope azure.ClusterScoper
+	switch cluster.Spec.InfrastructureRef.Kind {
```

Comment:
Is there a decent way to extract this "get cluster scoper" logic into a function that can be resued?

Reply:
yes! thanks for the suggestion. Refactor incoming.

Reply:
Done! this helped a lot